ksql-engine/src/main/java/io/confluent/ksql/internal/KsqlEngineMetrics.java

```diff
@@ -208,71 +199,133 @@ public double measure(final MetricConfig metricConfig, final long l) {
           public void record(final MetricConfig metricConfig, final double v, final long l) {
             // We don't want to record anything, since the live queries anyway.
           }
-        });
+        }
+    );
+  }
 
-    sensor.add(
-        metrics.metricName(ksqlServiceId + "num-persistent-queries", this.metricGroupName,
-            "The current number of persistent queries running in this engine"),
-        new MeasurableStat() {
+  private void configureNumPersistentQueries(final Metrics metrics) {
+    final String metricName = "num-persistent-queries";
+    final String description = "The current number of persistent queries running in this engine";
+    createSensor(
+        metrics,
+        metricName,
+        description,
+        () -> new MeasurableStat() {
           @Override
           public double measure(final MetricConfig metricConfig, final long l) {
             return ksqlEngine.numberOfPersistentQueries();
           }
 
           @Override
           public void record(final MetricConfig metricConfig, final double v, final long l) {
-            // No action for record since we can read the desired results directly.
+            // We don't want to record anything, since the live queries anyway.
```

Comment:
nit: having trouble understanding this comment (though I see it's used elsewhere in the file as well...)

Is it saying that there's no state to record since we can always measure the value live? (Having trouble looking up how the record() method is used.)

Reply:
From the perspective of this diff that's just copypasta. I think what its trying to say is that we aren't using `record` to track the number of queries - the engine keeps track of it internally.